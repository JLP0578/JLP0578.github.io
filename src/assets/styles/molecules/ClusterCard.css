
.cluster_cards {
    display: grid;
    grid-gap: 4px;
    grid-template-columns: 1fr;
    grid-template-rows: 212px repeat(4,minmax(110px,auto));
    list-style: none;
    margin: 0;
    padding: 0;
}
.cluster_cards li:first-child {
    grid-column: 1;
    grid-row: 1;
}
@media screen and (min-width: 767px) {
    .cluster_cards {
        grid-gap: 23px;
        grid-template-columns: 1fr 1fr 1fr;
        grid-template-rows: repeat(2,minmax(198px,auto));
        padding: 0 44px;
    }

    .cluster_cards li {
        box-sizing: border-box;
        position: relative;
    }

    .cluster_cards li:first-child {
        grid-row: 1/span 2;
    }
}