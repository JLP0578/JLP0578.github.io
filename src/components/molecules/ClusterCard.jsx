import React from "react";
import PropTypes from 'prop-types';

import Card from "../atoms/Card";

import '../../assets/styles/molecules/ClusterCard.css';

export default class ClusterCard extends React.Component {
    
    render() {
        return (
            <>
            <br/>
            <ul class="cluster_cards">
                <li><Card 
                    template="bizLT"
                    title="title"
                    content="content"
                    imageAlt="z"
                    imageUrl="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
                 /></li>
                <li><Card 
                    template="bizN"
                    title="title"
                    content="content"
                    imageAlt="z"
                    imageUrl="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
                 /></li>
                <li><Card 
                    template="bizN"
                    title="title"
                    content="content"
                    imageAlt="z"
                    imageUrl="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
                 /></li>
                <li><Card 
                    template="bizN"
                    title="title"
                    content="content"
                    imageAlt="z"
                    imageUrl="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
                 /></li>
                <li><Card 
                    template="bizRB"
                    title="title"
                    content="content"
                    imageAlt="z"
                    imageUrl="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
                 /></li>
            </ul>
            </>
        ); 
    }
}

ClusterCard.propTypes = {
    imageAlt: PropTypes.string,
    imageUrl: PropTypes.string,
    title: PropTypes.string,
    content: PropTypes.string,
    buttonAlt: PropTypes.string,
    buttonUrl: PropTypes.string,
    buttonContent: PropTypes.string,
};

ClusterCard.defaultProps = {
};




