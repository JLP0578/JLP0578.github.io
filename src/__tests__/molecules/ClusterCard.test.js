import renderer from 'react-test-renderer';

import ClusterCard from '../../components/molecules/ClusterCard';

test('renders correctly', () => {
  const tree = renderer
    .create(<ClusterCard />)
    .toJSON();

  expect(tree).toMatchSnapshot();

});