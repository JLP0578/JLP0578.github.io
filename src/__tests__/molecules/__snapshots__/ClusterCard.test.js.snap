// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <br />,
  <ul
    class="cluster_cards"
  >
    <li>
      <div
        className="bizLT card card__outer"
      >
        <div
          className="card card__alf"
        >
          <img
            alt="z"
            className="card__img"
            src="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
          />
          <h1
            className="card__title"
          >
            title
          </h1>
          <div
            className="card card__inner"
          >
            <p>
              content
            </p>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        className="bizN card card__outer"
      >
        <div
          className="card card__alf"
        >
          <img
            alt="z"
            className="card__img"
            src="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
          />
          <h1
            className="card__title"
          >
            title
          </h1>
          <div
            className="card card__inner"
          >
            <p>
              content
            </p>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        className="bizN card card__outer"
      >
        <div
          className="card card__alf"
        >
          <img
            alt="z"
            className="card__img"
            src="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
          />
          <h1
            className="card__title"
          >
            title
          </h1>
          <div
            className="card card__inner"
          >
            <p>
              content
            </p>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        className="bizN card card__outer"
      >
        <div
          className="card card__alf"
        >
          <img
            alt="z"
            className="card__img"
            src="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
          />
          <h1
            className="card__title"
          >
            title
          </h1>
          <div
            className="card card__inner"
          >
            <p>
              content
            </p>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        className="bizRB card card__outer"
      >
        <div
          className="card card__alf"
        >
          <img
            alt="z"
            className="card__img"
            src="https://www.cssscript.com/demo/cyberpunk-2077/img/tile2.jpg"
          />
          <h1
            className="card__title"
          >
            title
          </h1>
          <div
            className="card card__inner"
          >
            <p>
              content
            </p>
          </div>
        </div>
      </div>
    </li>
  </ul>,
]
`;
